packages/media-query/src/media-query.hook.tsx

```diff
@@ -148,3 +150,82 @@ const createMediaQueries = (breakpoints: Dict) => {
     return mediaQuery
   })
 }
+
+type BreakpointValues<T> = { [breakpoint: string]: T }
+
+/**
+ * React hook for getting the value for the current breakpoint from the
+ * provided responsive values object.
+ *
+ * @example
+ *
+ * const width = useBreakpointValue({ base: '150px', md: '250px' })
+ * return <Button width={width}>I'm {width} wide!</Button>
+ */
+export function useBreakpointValue<T = any>(values: BreakpointValues<T>) {
+  const breakpoint = useBreakpoint()
+
+  // if breakpoint is undefined, we'll return undefined
+  if (!breakpoint) return
+
+  // if we have no value for the current breakpoint, return the `base`
+  // breakpoint value
+  const baseValue = values.base
+
+  return values[breakpoint] || baseValue
+}
+
+/**
+ * React hook for getting the user's animation preference. Returns `false` if
+ * user prefers reduced motion.
+ *
+ * Inspired by @kripod:
+ * https://github.com/kripod/react-hooks/blob/10f1b489078a6c61bd2226258feef1d2ced915a2/packages/web-api-hooks/src/useMotionPreference.ts
+ */
+export function useAnimationPreference() {
+  const isReduce = useMediaQuery("(prefers-reduced-motion: reduce)")
+  return !isReduce
+}
+
+/**
+ * React hook for getting the user's color mode preference. Returns `"light"`,
+ * `"dark"`, or `undefined` if no preference exists.
+ *
+ * Inspired by @kripod:
+ * https://github.com/kripod/react-hooks/blob/10f1b489078a6c61bd2226258feef1d2ced915a2/packages/web-api-hooks/src/useColorSchemePreference.ts
+ */
+export function useColorModePreference() {
+  const isLight = useMediaQuery("(prefers-color-scheme: light)")
+  const isDark = useMediaQuery("(prefers-color-scheme: dark)")
+
+  if (isLight) return "light"
+  if (isDark) return "dark"
+}
+
+/**
+ * React hook for getting the width and height of a component.
+ *
+ * Inspired by @kripod:
+ * https://github.com/kripod/react-hooks/blob/10f1b489078a6c61bd2226258feef1d2ced915a2/packages/web-api-hooks/src/useSize.ts
+ */
+export function useViewportSize(
+  ref: React.RefObject<HTMLElement>,
+): Readonly<[number, number]> {
+  const [size, setSize] = useState<Readonly<[number, number]>>([0, 0])
+
+  useEffect(() => {
+    if (!ref.current) return undefined
+
+    const observer = new ResizeObserver(([entry]) => {
+      const { inlineSize: width, blockSize: height } = entry.contentBoxSize[0]
```

Comment:
`ResizeObserver` gives us `contentBoxSize` or `borderBoxSize`, and I'm not sure which would be more appropriate here. Should this include the padding/border? Or maybe we should return the width/height for both and let the user decide?

Reply:
Should we consider using "@reach/rect" for this part? It's much smaller than the resize observer package and works really well.

What do you think? I feel we can simply re-export `useRect` from `@reach/rect`

Reply:
That's a better idea than using the larger package. Though maybe the best idea is not even re-exporting and letting users decide which to use if they want this functionality?